### PR TITLE
Compute Validity period

### PR DESCRIPTION
**Pull request by datanel**

### CODE REVIEW

File: src/gtfs/read.rs [hunk 1]
```diff
@@ -957,6 +941,42 @@ pub fn read_routes<P: AsRef<path::Path>>(path: P, collections: &mut Collections)
...
+        acc.union(&c.dates).cloned().collect()
+    });
+
+    if dates.is_empty() {
+        return None;
+    }
+
+    Some(objects::ValidityPeriod {
+        start_date: *dates.iter().next().unwrap(),
+        end_date: *dates.iter().next_back().unwrap(),
```
Comment (TeXitoi) [inline src/gtfs/read.rs]: It will fail with only one day.

Comment (datanel) [reply src/gtfs/read.rs]: no it won't. I added a test for this case

Comment (TeXitoi) [reply src/gtfs/read.rs]: ha, my bad, was thinking it was the same iterator, but there is one for each.

Thus, what's wrong with front() and back()?

Comment (TeXitoi) [reply src/gtfs/read.rs]: OK, my bad.

**Pull request merged**